src/resolve/resolved-dns-transaction.h

```diff
@@ -1,12 +1,24 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 #pragma once
 
+
 #include "list.h"
 #include "resolved-def.h"
 #include "resolved-dns-dnssec.h"
 #include "resolved-dns-server.h"
 #include "resolved-forward.h"
 
+#if ENABLE_DNS_OVER_HTTPS
+#include <curl/curl.h>
+#endif
+
+#if ENABLE_DNS_OVER_HTTPS
+#include "curl-util.h"
+#endif
```

Comment:
please make it so that curl-util.h can always be included, and thus move the ifdeffery into the header itself. We really prefer if ifdeffery is minimized in consuming code and always placed in the  providing code, if you follow what i mean